stacks_queues/queue.py

```diff
@@ -1,61 +1,91 @@
 
 INITIAL_QUEUE_SIZE = 20
 
+
 class QueueFullException(Exception):
     pass
 
+
 class QueueEmptyException(Exception):
     pass
 
+
 class Queue:
 
     def __init__(self):
         self.store = [None] * INITIAL_QUEUE_SIZE
         self.buffer_size = INITIAL_QUEUE_SIZE
-        self.front = -1
-        self.rear = -1
+        self.front = 0
+        self.rear = 0
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        # if self.front == (self.rear + 1) % self.buffer_size:
+        if self.size == self.buffer_size:
+            raise(QueueFullException)
+        if self.rear == self.buffer_size and self.front > 0:
+            self.rear = 0
+        self.store[self.rear] = element
+        self.rear += 1
+        self.size += 1
+        return None
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.front == self.rear:
+            raise(QueueEmptyException)
+        temp = self.store[self.front]
+        self.store[self.front] = None
+        self.front += 1
+        self.size -= 1
+        return temp
```

Comment:
You should also have the code for this method make the queue wrap around the buffer.

```suggestion
        if self.front == self.rear:
            raise(QueueEmptyException)
        temp = self.store[self.front]
        self.store[self.front] = None
        # This will make the front wrap around to 0 if you go past the end of the buffer indices.
        self.front = (self.front + 1) % self.buffer_size
        self.size -= 1
        return temp
```